### ISSUE TITLE
... is not responding.

### ISSUE THREAD

**Issue by SzilagyiR** [NorthernMan54/homebridge-alexa#34]
Hi!

I’am using homebridge with alexa several month without any issue.
About 3-4 weeks a go she answers my ask “xy not responding” when I repeat it she turns the swich.

Have you got any idea what is her problem?

Thanks!

**Comment by NorthernMan54** [comment]
I have seen this occasionally with mine, it typically occurs if a device is slow to respond to a command.  But in my setup it happens about once a month randomly, so I never really looked into in.  If you can recreate the issue, and have debug loging enabled.  A log fragment when it occurs would be interesting to look at.  I expect the homebridge is just slow responding to the command, causing the not responding message.

**Comment by SzilagyiR** [comment]
The issue in my setup is constant now. I will try to turn on debuge mode and save the log, and post it here.

Is it possible to extend the wait time? It is possible to my RaspberryPi is slower now than before?

**Comment by SzilagyiR** [comment]
I tried the debuge mode, but nothing happens when Alexa not respond.

**Comment by NorthernMan54** [comment]
@SzilagyiR Low and behold, I found that one of my lights always gets a 'Not responding'.  And after looking at it further I found that the light was taking > 5 seconds to perform the command.  And that Alexa was timing out before the command could complete.  It looks like the problem is devices not responding fast enough for her.

Does this feel like what you are seeing?

**Comment by SzilagyiR** [comment]
Hi!

My Alexa's first respons still "xy not responding”, second try works perfectly. I tried everything but nothing changed...

I think too to response time is the blame. Where can I looking after my switches, about respons time? I do not understand why worked without any issue till november, and why not now...

**Comment by NorthernMan54** [comment]
If have one lamp in my setup that does the same not responding when I ask it to change to brightness.   When I look at the log, I can see that the device is taking 7 seconds to complete the action.  It first gets the on/off status of the light, then it changes the setting.  Between the two commands 7 seconds, and at about 5 seconds Alexa starts saying not responding.

This log is from home bridge running is debug mode ie

DEBUG=* homebridge


Jan 11 18:27:28 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:28 GMT EventedHTTPServer [::ffff:192.168.1.206] HTTP request: /api/pkBovaHrTjmiUwGqH7m6q7eecgD3VXVrR3BPE1qK/lights/6
Jan 11 18:27:28 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:28 GMT HAPServer [CC:22:3D:E3:CE:30] HAP Request: GET /api/pkBovaHrTjmiUwGqH7m6q7eecgD3VXVrR3BPE1qK/lights/6
Jan 11 18:27:28 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:28 GMT Accessory [Penny] Getting value for Characteristic "On"
Jan 11 18:27:28 penny homebridge[32351]: [1/11/2018, 6:27:28 PM] [Heyu] Got power state of A6 1
Jan 11 18:27:28 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:28 GMT Accessory [Penny] Got Characteristic "On" value: true
Jan 11 18:27:28 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:28 GMT hue light [ { aid: 7, iid: 10, value: true, status: 0 } ]
Jan 11 18:27:28 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:28 GMT EventedHTTPServer [::ffff:192.168.1.206] HTTP Response is finished
Jan 11 18:27:29 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:29 GMT EventedHTTPServer [::ffff:192.168.1.206] HTTP request: /api/pkBovaHrTjmiUwGqH7m6q7eecgD3VXVrR3BPE1qK/lights/6/state
Jan 11 18:27:29 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:29 GMT HAPServer [CC:22:3D:E3:CE:30] HAP Request: PUT /api/pkBovaHrTjmiUwGqH7m6q7eecgD3VXVrR3BPE1qK/lights/6/state
Jan 11 18:27:29 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:29 GMT Accessory [Penny] Processing characteristic set: [{"aid":7,"iid":10,"value":1},{"aid":7,"iid":11,"value":20}]
Jan 11 18:27:29 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:29 GMT Accessory [Penny] Setting Characteristic "On" to value 1
Jan 11 18:27:29 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:29 GMT Accessory [Penny] Setting Characteristic "Brightness" to value 20
Jan 11 18:27:30 penny homebridge[32351]: [1/11/2018, 6:27:30 PM] [Heyu] Set power state of A6 to on
Jan 11 18:27:30 penny homebridge[32351]: [1/11/2018, 6:27:30 PM] [Heyu] Got brightness level of A6 100
Jan 11 18:27:35 penny homebridge[32351]: [1/11/2018, 6:27:35 PM] [Heyu] Set Bright/Dim dim A6 18 ( 18.16 % )
Jan 11 18:27:35 penny homebridge[32351]: Thu, 11 Jan 2018 23:27:35 GMT EventedHTTPServer [::ffff:192.168.1.206] HTTP Response is finished
Jan 11 18:27:35 penny homebridge[32351]: [1/11/2018, 6:27:35 PM] [Heyu] Got brightness level of A6 6
Jan 11 18:27:35 penny homebridge[32351]: [1/11/2018, 6:27:35 PM] [Heyu] Got power state of A6 1

**Comment by sejmann** [comment]
This started happening to me tonight, only after upgrading to nodejs 8.9.4 from 7.something.  I'll have to troubleshoot.

**Comment by NorthernMan54** [comment]
Npm version 5 is problematic.  It has known issues with installing code direct from github.  I’m on the version 7 stream of nodejs, and version 3 of npm
